src/test/java/kdt/springbootjpa/customer/CustomerRepositoryTest.java

```diff
@@ -0,0 +1,64 @@
+package kdt.springbootjpa.customer;
+
+import kdt.springbootjpa.customer.entity.Customer;
+import kdt.springbootjpa.customer.repository.CustomerRepository;
+import org.junit.jupiter.api.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@DataJpaTest
+public class CustomerRepositoryTest {
+
+    @Autowired
+    private CustomerRepository customerRepository;
+
+    @Test
+    @DisplayName("[성공] Customer 저장 및 조회하기")
+    void testSaveAndGetCustomer() {
+        Customer customer = Customer.builder().firstName("tlsdud").lastName("fortune").build();
+        customerRepository.save(customer);
+
+        Customer savedCustomer = customerRepository.findById(customer.getId()).get();
+
+        assertThat(savedCustomer.getFirstName()).isEqualTo(customer.getFirstName());
+        assertThat(savedCustomer.getLastName()).isEqualTo(customer.getLastName());
+        List<Customer> customers = customerRepository.findAll();
+        assertThat(customers.size()).isEqualTo(1);
+    }
+
+    @Test
+    @DisplayName("[성공] Customer 수정하기")
+    void testUpdateCustomer() {
+        Customer customer = Customer.builder().firstName("sinyoung").lastName("bok").build();
+        customerRepository.save(customer);
+
+        String newFirstName = "tlsdud";
+        String newLastName = "fortune";
+        Customer savedCustomer = customerRepository.findById(customer.getId()).get();
+        savedCustomer.changeFirstName(newFirstName);
+        savedCustomer.changeLastName(newLastName);
+
+        Customer updatedCustomer = customerRepository.findById(customer.getId()).get();
+        assertThat(updatedCustomer.getFirstName()).isEqualTo(newFirstName);
+        assertThat(updatedCustomer.getLastName()).isEqualTo(newLastName);
```

Comment:
```suggestion

        Optional<Customer> updatedCustomer = customerRepository.findById(customer.getId());

        assertThat(updatedCustomer)
            .hasValueSatisfying(customer1 -> assertAll(
                () -> assertThat(customer1.getFirstName()).isEqualTo(newFirstName),
                () -> assertThat(customer1.getLastName()).isEqualTo(newLastName)
            ));
```
신영님께서 커멘트로 공유주신 사이트 내용에 따르면 Optional 같은 경우에는 위와 같은 코드 방식으로 테스트 할 수 있을 것 같습니다.
.get() 차제를 호출하는 것은 엄밀하게 이야기 하면 Optional을 사용 안하는 것과 다름 없다고 생각합니다.
개인적으로 Optional get() 호출 안하고 어떻게 해야 할까 고민했었습니다. 덕분에 저도 배워갑니다.!

Reply:
오호 참고하겠습니다!
추가로 바로 이전 코드에서의 get 메소드는 어떻게 처리하면 좋을까요?


`Customer savedCustomer = customerRepository.findById(customer.getId()).get();`
`savedCustomer.changeFirstName(newFirstName);`
`savedCustomer.changeLastName(newLastName);`

Reply:
get()이라는 것을 없애고 싶으신건가요? 테스트 코드 문맥 상 이 친구는 assertThat()으로 검증할 필요는 없는 내용으로 보여서
만약 get()을 그냥 없애기만 하면된다면 ifPresent() api도 존재합니다.

```java
        customerRepository.findById(customer.getId())
	        .ifPresent(c -> {
	            c.changeFirstName(newFirstName);
	            c.changeLastName(newLastName);
	        });

```